packages/gatsby-admin/src/utils/use-develop-logs.ts

```diff
@@ -0,0 +1,65 @@
+import React from "react"
+import io from "socket.io-client"
+import { useServices } from "../components/services-provider"
+
+type RestartState = "idle" | "needs-restart" | "is-restarting"
+
+type RestartFn = () => void
+
+function useDevelopState(): [RestartState, RestartFn] {
+  const [state, setState] = React.useState<RestartState>(`idle`)
+  const [socket, setSocket] = React.useState<ReturnType<typeof io> | null>(null)
```

Comment:
If you're interested in putting the `socket` outside the component `state`, I may have a trick 😊 https://github.com/mfrachet/use-socketio/blob/master/src/io/provider.tsx#L10

Reply:
We'll likely only need this socket in this one hook, but if we ever need to share it that solution is really great to have 😍👍